### PR TITLE
refactor: polymarket provider supports both /price and /midpoint

### PR DESCRIPTION
**Pull request by technicallyty**
request ala dydx 

https://skip-protocol.slack.com/archives/C050VLDNSSV/p1723058836493679?thread_ts=1723057286.702739&cid=C050VLDNSSV

### CODE REVIEW

File: providers/apis/polymarket/README.md [hunk 1]
```diff
@@ -16,12 +16,9 @@ The offchain ticker is expected to be _just_ the token_id.
 
 example: `95128817762909535143571435260705470642391662537976312011260538371392879420759`
 
-The Provider simply calls the `/price` endpoint of the CLOB API. There are two query parameters:
+The Provider simply calls the `/midpoint` endpoint of the CLOB API. There are two query parameters:
```
Comment (aljo242) [inline providers/apis/polymarket/README.md]: looks like there is 1 query parameter now

**Review by aljo242 — APPROVED** [review]
lgtm w one small nit

**Comment by chenyaoy** [comment]
Will this support both `https://clob.polymarket.com/price?token_id=%s&side=...` and midpoint?

**Comment by technicallyty** [comment]
> Will this support both `https://clob.polymarket.com/price?token_id=%s&side=...` and midpoint?

nope just switched to midpoint - i can add support for both

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**